src/lib/holocene/toggle-switch.svelte

```diff
@@ -1,53 +1,130 @@
   export let disabled = false;
+  export let label: string;
+  export let labelPosition: 'left' | 'right' = 'right';
+  export let labelHidden = false;
+
+  let focused = false;
+  let hovered = false;
+
+  const handleFocus = () => {
+    focused = true;
+  };
+
+  const handleBlur = () => {
+    focused = false;
+  };
+
+  const handleMouseEnter = () => {
+    hovered = true;
+  };
+
+  const handleMouseLeave = () => {
+    hovered = false;
+  };
 </script>
 
-<label for={id} class="switch">
+<label class="switch" class:disabled>
+  {#if labelPosition === 'left'}
+    <span class="label left" class:sr-only={labelHidden}>
+      {label}
+    </span>
+  {/if}
   <input
     on:change
     {id}
     {disabled}
     bind:checked
     type="checkbox"
-    class="hidden"
-    class:disabled
+    class="checkbox"
+    role="switch"
+    on:focus={handleFocus}
+    on:blur={handleBlur}
     data-testid={$$props['data-testid']}
   />
-  <span class="slider" />
+  <span
+    on:mouseenter={handleMouseEnter}
+    on:mouseleave={handleMouseLeave}
+    class="slider"
+    class:disabled
+    class:checked
+    class:focused
+    class:hovered
+  >
+    <span class="thumb" class:checked />
+  </span>
+  {#if labelPosition === 'right'}
+    <span class="label right" class:sr-only={labelHidden}>
+      {label}
+    </span>
+  {/if}
 </label>
 
 <style lang="postcss">
   .switch {
-    @apply relative inline-block h-8 w-[52px] rounded-[8rem];
+    @apply px-2 relative flex items-center rounded max-w-fit cursor-pointer;
+
+    &.disabled {
+      @apply pointer-events-none;
+    }
   }
 
-  .slider {
-    @apply absolute top-0 left-0 right-0 bottom-0 cursor-pointer rounded-[8rem] border-2 border-gray-600 bg-gray-100 duration-300;
+  .label {
+    @apply whitespace-nowrap text-sm;
 
-    &::before {
-      @apply absolute left-1 bottom-0.5 h-6 w-6 rounded-[50%] bg-gray-900 duration-300;
+    &.left {
+      @apply mr-2;
+    }
 
-      content: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M16.8438 16.875C16.6562 17.0625 16.3125 17.0625 16.125 16.875L12 12.7188L7.84375 16.875C7.65625 17.0625 7.3125 17.0625 7.125 16.875C6.9375 16.6875 6.9375 16.3438 7.125 16.1562L11.2812 12L7.125 7.875C6.9375 7.6875 6.9375 7.34375 7.125 7.15625C7.3125 6.96875 7.65625 6.96875 7.84375 7.15625L12 11.3125L16.125 7.15625C16.3125 6.96875 16.6562 6.96875 16.8438 7.15625C17.0312 7.34375 17.0312 7.6875 16.8438 7.875L12.6875 12L16.8438 16.1562C17.0312 16.3438 17.0312 16.6875 16.8438 16.875Z' fill='%23fff'%3E%3C/path%3E%3C/svg%3E");
+    &.right {
+      @apply ml-2;
     }
   }
 
-  input + .slider {
-    @apply bg-white;
+  .checkbox {
+    @apply absolute w-0 opacity-0;
   }
 
-  input.disabled + .slider {
-    @apply bg-gray-200 cursor-not-allowed;
+  .slider {
+    @apply flex relative items-center h-5 w-9 rounded-xl border border-gray-600 bg-gray-50;
+
+    &.hovered {
+      @apply bg-gray-200;
```

Comment:
What do you think about adding a hover state for the toggled on state as well? Maybe a slightly darker indigo?

<img width="83" alt="Screenshot 2023-08-02 at 9 23 32 AM" src="https://github.com/temporalio/ui/assets/15069288/838a83f6-bf6e-438c-ba71-4a5482016d2c">

Reply:
I'll check with Ash, but the hover state in Figma is the same as the default checked state.
<img width="541" alt="Screen Shot 2023-08-02 at 3 44 06 PM" src="https://github.com/temporalio/ui/assets/11775628/dc828a78-62ab-4051-b2ed-0b8f3065667b">